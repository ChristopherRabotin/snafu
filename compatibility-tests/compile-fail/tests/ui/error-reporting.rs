mod wrong_error_types {
    use snafu::Snafu;

    #[derive(Snafu)]
    union AUnion {
        _a: i32,
    }

    #[derive(Snafu)]
    enum TupleEnumVariant {
        Alpha(i32),
    }
}

mod other_attributes {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    enum Error {
        #[serde]
        UnknownVariantAttributeIsIgnored,
    }
}

mod display {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    enum FailedAttributeParsing {
        #[snafu(display)]
        DisplayWithoutArgument,
    }

    #[derive(Debug, Snafu)]
    enum InvalidGeneratedCode {
        #[snafu(display(foo()))]
        FormatStringMissing,

        #[snafu(display(42))]
        FormatStringNotStringLiteral,
    }
}

mod opaque {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    struct UnitStruct;

    #[derive(Debug, Snafu)]
    struct NamedFieldStruct {
        some_field: i32,
    }

    #[derive(Debug, Snafu)]
    struct ShortTupleStruct();

    #[derive(Debug, Snafu)]
    struct LongTupleStruct(i32, i32);
}

fn main() {}
