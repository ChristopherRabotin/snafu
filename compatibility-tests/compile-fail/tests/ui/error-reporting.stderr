error: Can only derive `Snafu` for an enum or a newtype
 --> $DIR/error-reporting.rs:5:5
  |
5 |     union AUnion {
  |     ^^^^^

error: Can only derive `Snafu` for enums with struct-like and unit enum variants
  --> $DIR/error-reporting.rs:11:14
   |
11 |         Alpha(i32),
   |              ^^^^^

error: unexpected end of input, expected parentheses
  --> $DIR/error-reporting.rs:30:16
   |
30 |         #[snafu(display)]
   |                ^^^^^^^^^

error: format argument must be a string literal
  --> $DIR/error-reporting.rs:36:25
   |
36 |         #[snafu(display(foo()))]
   |                         ^^^^^
   |
help: you might be missing a string literal to format with
   |
36 |         #[snafu(display("{}", foo()))]
   |                         ^^^^^

error: format argument must be a string literal
  --> $DIR/error-reporting.rs:39:25
   |
39 |         #[snafu(display(42))]
   |                         ^^
   |
help: you might be missing a string literal to format with
   |
39 |         #[snafu(display("{}", 42))]
   |                         ^^^^^

error: Can only derive `Snafu` for tuple structs with exactly one field
  --> $DIR/error-reporting.rs:56:5
   |
56 |     struct ShortTupleStruct();
   |     ^^^^^^

error: Can only derive `Snafu` for tuple structs with exactly one field
  --> $DIR/error-reporting.rs:59:5
   |
59 |     struct LongTupleStruct(i32, i32);
   |     ^^^^^^

error: cannot find attribute `serde` in this scope
  --> $DIR/error-reporting.rs:20:11
   |
20 |         #[serde]
   |           ^^^^^
